src/LinuxTracing/CMakeLists.txt

```diff
@@ -96,3 +96,12 @@ target_link_libraries(LinuxTracingTests PRIVATE
         GTest::Main)
 
 register_test(LinuxTracingTests)
+
+add_executable(ThreadDepedencyTestApp)
```

Comment:
Generally speaking, I'm not a big fan of test binaries in the codebase, because:
- if it becomes a practice, they can grow out of control;
- they are immediately dead code.

It's fine for this time but I'm going to ask you to create a separate pull request for it, which will result in a separate commit in the history.